mtp_api/apps/mtp_auth/tests/test_views.py

```diff
@@ -74,30 +79,332 @@ def test_can_access_my_data_including_managing_prisons(self):
 
     def test_correct_permissions_returned(self):
         for user in self.test_users:
-            self.client.force_authenticate(user=user)
-
             url = self._get_url(user.username)
-            response = self.client.get(url, format='json')
+            response = self.client.get(
+                url, format='json',
+                HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+            )
             self.assertEqual(response.status_code, status.HTTP_200_OK)
 
             self.assertEqual(response.data['permissions'], user.get_all_permissions())
 
     def test_my_data_with_empty_prisons(self):
-        users = \
-            self.prisoner_location_admins + \
+        users = (
+            self.prisoner_location_admins +
             self.bank_admins + self.refund_bank_admins
+        )
 
         for user in users:
-            self.client.force_authenticate(user=user)
-
             url = self._get_url(user.username)
-            response = self.client.get(url, format='json')
+            response = self.client.get(
+                url, format='json',
+                HTTP_AUTHORIZATION=self.get_http_authorization_for_user(user)
+            )
             self.assertEqual(response.status_code, status.HTTP_200_OK)
 
             self.assertEqual(response.data['pk'], user.pk)
             self.assertEqual(response.data['prisons'], [])
 
 
+class ListUserTestCase(APITestCase, AuthTestCaseMixin):
+    fixtures = [
+        'initial_groups.json',
+        'test_prisons.json'
+    ]
+
+    def setUp(self):
+        super().setUp()
+        (
+            self.prison_clerks, self.prisoner_location_admins,
+            self.bank_admins, self.refund_bank_admins, _
+        ) = make_test_users(clerks_per_prison=1)
+        self.cashbook_uas, self.pla_uas, self.bank_uas = make_test_user_admins()
+
+    def get_url(self):
+        return reverse('user-list')
+
+    def test_list_users_only_accessible_to_admins(self):
+        response = self.client.get(
+            self.get_url(),
+            format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(self.refund_bank_admins[0])
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def _check_list_users_succeeds(self, requester, client_id):
+        response = self.client.get(
+            self.get_url(),
+            format='json',
+            HTTP_AUTHORIZATION=self.get_http_authorization_for_user(requester)
+        )
+
+        for user_item in response.data['results']:
+            user = User.objects.get(username=user_item['username'])
+            self.assertEqual(
+                user.applicationusermapping_set.all()[0].application.client_id,
```

Comment:
not really an issue in this test, but it's possible that some users will be accessing multiple applications (e.g. sharon going into bank and prisoner admins) so it doesn't have to be the first that matches.

ps: you can do `.first()` instead of `.all()[0]`